github-runners/terraform/base/network.tf

```diff
@@ -0,0 +1,185 @@
+
+  tags = {
+    Name = "Public Subnet ${count.index + 1}"
+  }
+}
+
+resource "aws_subnet" "private_subnets" {
+  count             = length(var.private_subnet_cidrs)
+  vpc_id            = aws_vpc.gh-runners.id
+  cidr_block        = element(var.private_subnet_cidrs, count.index)
+  availability_zone = element(var.azs, count.index)
+
+  tags = {
+    Name = "Private Subnet ${count.index + 1}"
+  }
+
+  lifecycle {
+    prevent_destroy = false
+  }
+
+}
+
+
+resource "aws_internet_gateway" "gw" {
+  count  = length(var.public_subnet_cidrs)
+  vpc_id = aws_vpc.gh-runners.id
+
+  tags = {
+    Name = "igw-public-subnet-${count.index}"
+  }
+
+  lifecycle {
+    prevent_destroy = false
+  }
+}
+
+resource "aws_route_table" "public_route_table" {
+  count  = length(var.public_subnet_cidrs)
+  vpc_id = aws_vpc.gh-runners.id
+
+  route {
+    cidr_block = "0.0.0.0/0"
+    gateway_id = aws_internet_gateway.gw[count.index].id
+  }
+
+  route {
+    ipv6_cidr_block = "::/0"
+    gateway_id      = aws_internet_gateway.gw[count.index].id
+  }
+
+  tags = {
+    Name = "public-route-tbl-${count.index}"
+  }
+
+  depends_on = [
+    aws_internet_gateway.gw
+  ]
+}
+
+resource "aws_route_table_association" "public_route_table_subnets_association" {
+  count          = length(var.public_subnet_cidrs)
+  subnet_id      = element(aws_subnet.public_subnets.*.id, count.index)
+  route_table_id = element(aws_route_table.public_route_table.*.id, count.index)
+}
+
+
+resource "aws_route_table" "private_route_table" {
+  count  = length(var.private_subnet_cidrs)
+  vpc_id = aws_vpc.gh-runners.id
+
+  route {
+    cidr_block     = "0.0.0.0/0"
+    nat_gateway_id = aws_nat_gateway.nat_gateway[count.index].id
+  }
+
+  tags = {
+    Name = "private-route-tbl-${count.index}"
+  }
+}
+
+resource "aws_route_table_association" "private_route_table_subnets_association" {
+  count          = length(var.private_subnet_cidrs)
+  subnet_id      = element(aws_subnet.private_subnets.*.id, count.index)
+  route_table_id = element(aws_route_table.private_route_table.*.id, count.index)
+}
+
+resource "aws_security_group" "allow_runner" {
+  name        = "allow_runner"
+  description = "Allow HTTP and HTTPS inbound traffic"
+  vpc_id      = aws_vpc.gh-runners.id
+
+  ingress {
+    description = "HTTPS for VPC"
+    from_port   = 443
+    to_port     = 443
+    protocol    = "tcp"
+    cidr_blocks = [aws_vpc.gh-runners.cidr_block]
+  }
+
+  ingress {
+    description = "HTTP for VPC"
+    from_port   = 80
+    to_port     = 80
+    protocol    = "tcp"
+    cidr_blocks = [aws_vpc.gh-runners.cidr_block]
+  }
+
+  ingress {
+    description = "SSH for VPC"
+    from_port   = 22
+    to_port     = 22
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  ingress {
+    description = "WinRM for VPC"
+    from_port   = 5985
+    to_port     = 5985
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  ingress {
+    description = "RDP for VPC"
```

Comment:
why do we need a RDP here and WinRm above?

Reply:
RDP is for remote access using SSM agent or a RDP client. The WinRM is used for deployments similar to SSH.